Script Analysis Examples/Test Data Analysis/Failure_Pareto_Result_Analysis.ipynb

```diff
@@ -0,0 +1,642 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Handle grouping by day\n",
+    "If the grouping is by day, the group name is the date and time when the test started in UTC. To group all test results from a single day together, convert to server time and remove time information from the group name."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_results_copy = copy.copy(df_results)\n",
+    "df_results_copy.fillna(value='', inplace=True)\n",
+    "\n",
+    "if grouping == 'started_at':\n",
+    "    truncated_times = []\n",
+    "    for val in df_results_copy[grouping]:\n",
+    "        local_time = val.astimezone(tz.tzlocal())\n",
+    "        truncated_times.append(str(datetime.date(local_time.year, local_time.month, local_time.day)))\n",
+    "    df_results_copy[grouping] = truncated_times"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Aggregate results into groups\n",
+    "Aggregate the data for each unique group and status.\n",
+    "\n",
+    "*See documentation for [size](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.size.html) and [unstack](https://pandas.pydata.org/pandas-docs/stable/generated/pandas.DataFrame.unstack.html) here.*"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_grouped = df_results_copy.groupby([grouping, 'status']).size().unstack(fill_value=0)\n",
+    "if 'PASSED' not in df_grouped:\n",
+    "    df_grouped['PASSED'] = 0\n",
+    "if 'FAILED' not in df_grouped:\n",
+    "    df_grouped['FAILED'] = 0\n",
+    "if 'ERRORED' not in df_grouped:\n",
+    "    df_grouped['ERRORED'] = 0"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Failure Pareto calculation\n",
+    "Count the number of test failures and calculate cumulative values for the pareto."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "df_fail_count = pd.DataFrame(df_grouped['FAILED'] + df_grouped['ERRORED'])\n",
+    "if grouping != 'started_at':\n",
+    "    df_fail_count.sort_values(by=[0], ascending=False, inplace=True)\n",
+    "total = df_fail_count[0].sum()\n",
+    "pareto_values = []\n",
+    "cumulative = 0\n",
+    "for data_member in df_fail_count[0]:\n",
+    "    cumulative += data_member\n",
+    "    pareto_values.append(100 * (cumulative / total))\n",
+    "\n",
+    "df_pareto = df_fail_count.reset_index().set_axis([grouping, 'fail_count'], axis=1)\n",
+    "df_pareto['cumulative'] = pareto_values\n",
+    "\n",
+    "if grouping == 'started_at':\n",
+    "    df_pareto['started_at'] = pd.to_datetime(df_pareto['started_at'])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
```

Comment:
We don't need this cell at all. The SystemLink reports output format isn't important in this example.

Reply:
removed the cell